### PR TITLE
Build: clean build warnings

### PR DESCRIPTION
**Pull request by mickeyouyou**
for DCHEKC_NOTNULL WARNING, use CHECK_NOTNULL don't need to change logic for now:
// You may see warnings in release mode if you don't use the return
// value of DCHECK_NOTNULL. Please just use DCHECK for such cases.

### CODE REVIEW

**Comment by mickeyouyou** [comment]
build log with warnings: 
[build_tmp.log](https://github.com/ApolloAuto/apollo/files/3987918/build_tmp.log)

File: modules/localization/ndt/ndt_localization.cc [hunk 1]
```diff
@@ -254,7 +254,7 @@ bool NDTLocalization::IsServiceStarted() { return is_service_started_; }
 
 void NDTLocalization::FillLocalizationMsgHeader(
     LocalizationEstimate* localization) {
-  DCHECK_NOTNULL(localization);
+  CHECK_NOTNULL(localization);
```
Comment (xiaoxq) [inline modules/localization/ndt/ndt_localization.cc]: 1. Use CHECK.
2. IMO, we can remove these CHECK on output pointers as it's pretty common sense.

File: modules/localization/rtk/rtk_localization_component.cc [hunk 2]
```diff
@@ -65,20 +65,20 @@ bool RTKLocalizationComponent::InitIO() {
   corrected_imu_listener_ = node_->CreateReader<localization::CorrectedImu>(
       imu_topic_, std::bind(&RTKLocalization::ImuCallback, localization_.get(),
                             std::placeholders::_1));
-  DCHECK_NOTNULL(corrected_imu_listener_);
+  CHECK_NOTNULL(corrected_imu_listener_);
```
Comment (xiaoxq) [inline modules/localization/rtk/rtk_localization_component.cc]: 1. Ditto, use CHECK
2. Or `corrected_imu_listener_ = CHECK_NOTNULL(node_->CreateReader(...));`

File: modules/localization/ndt/ndt_localization.cc (hunk 1 above)
Comment (mickeyouyou) [reply modules/localization/ndt/ndt_localization.cc]: Yes, you are right for output pointers check, I will remove it.

File: modules/planning/tasks/deciders/speed_decider/speed_decider.cc [hunk 3]
```diff
@@ -452,7 +452,7 @@ bool SpeedDecider::CreateYieldDecision(
 bool SpeedDecider::CreateOvertakeDecision(
     const Obstacle& obstacle,
     ObjectDecisionType* const overtake_decision) const {
-  DCHECK_NOTNULL(overtake_decision);
+  CHECK_NOTNULL(overtake_decision);
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/speed_decider/speed_decider.cc]: Also these ones plz.

File: modules/localization/rtk/rtk_localization.cc [hunk 4]
```diff
@@ -401,7 +399,7 @@ bool RTKLocalization::InterpolateIMU(const CorrectedImu &imu1,
                                      const CorrectedImu &imu2,
                                      const double timestamp_sec,
                                      CorrectedImu *imu_msg) {
-  DCHECK_NOTNULL(imu_msg);
+  CHECK_NOTNULL(imu_msg);
```
Comment (xiaoxq) [inline modules/localization/rtk/rtk_localization.cc]: ditto

File: modules/planning/common/path/path_data.cc [hunk 5]
```diff
@@ -119,8 +119,8 @@ common::PathPoint PathData::GetPathPointWithPathS(const double s) const {
 
 bool PathData::GetPathPointWithRefS(const double ref_s,
                                     common::PathPoint *const path_point) const {
-  DCHECK_NOTNULL(reference_line_);
-  DCHECK_NOTNULL(path_point);
+  CHECK_NOTNULL(reference_line_);
+  CHECK_NOTNULL(path_point);
```
Comment (xiaoxq) [inline modules/planning/common/path/path_data.cc]: ditto

File: modules/planning/common/path/path_data.cc [hunk 6]
```diff
@@ -177,7 +177,7 @@ std::string PathData::DebugString() const {
 
 bool PathData::SLToXY(const FrenetFramePath &frenet_path,
                       DiscretizedPath *const discretized_path) {
-  DCHECK_NOTNULL(discretized_path);
+  CHECK_NOTNULL(discretized_path);
```
Comment (xiaoxq) [inline modules/planning/common/path/path_data.cc]: ditto

File: modules/planning/navi/decider/navi_path_decider_test.cc [hunk 7]
```diff
@@ -42,7 +42,7 @@ class NaviPathDeciderTest : public ::testing::Test {
   static void GeneratePathData(
       double s, double init_y, double kappa,
       std::vector<common::PathPoint>* const path_points) {
-    DCHECK_NOTNULL(path_points);
+    CHECK_NOTNULL(path_points);
```
Comment (xiaoxq) [inline modules/planning/navi/decider/navi_path_decider_test.cc]: ditto

File: modules/planning/navi_planning.cc [hunk 8]
```diff
@@ -398,7 +398,7 @@ std::string NaviPlanning::GetCurrentLaneId() {
 
 void NaviPlanning::GetLeftNeighborLanesInfo(
     std::vector<std::pair<std::string, double>>* const lane_info_group) {
-  DCHECK_NOTNULL(lane_info_group);
+  CHECK_NOTNULL(lane_info_group);
```
Comment (xiaoxq) [inline modules/planning/navi_planning.cc]: ditto

File: modules/planning/navi_planning.cc [hunk 9]
```diff
@@ -426,7 +426,7 @@ void NaviPlanning::GetLeftNeighborLanesInfo(
 
 void NaviPlanning::GetRightNeighborLanesInfo(
     std::vector<std::pair<std::string, double>>* const lane_info_group) {
-  DCHECK_NOTNULL(lane_info_group);
+  CHECK_NOTNULL(lane_info_group);
```
Comment (xiaoxq) [inline modules/planning/navi_planning.cc]: ditto

File: modules/planning/planner/public_road/public_road_planner.cc [hunk 10]
```diff
@@ -33,7 +33,7 @@ Status PublicRoadPlanner::Init(const PlanningConfig& config) {
 Status PublicRoadPlanner::Plan(const TrajectoryPoint& planning_start_point,
                                Frame* frame,
                                ADCTrajectory* ptr_computed_trajectory) {
-  DCHECK_NOTNULL(frame);
+  CHECK_NOTNULL(frame);
```
Comment (xiaoxq) [inline modules/planning/planner/public_road/public_road_planner.cc]: ditto

File: modules/planning/tasks/deciders/path_decider/path_decider.cc [hunk 11]
```diff
@@ -66,7 +66,7 @@ Status PathDecider::Process(const ReferenceLineInfo *reference_line_info,
 bool PathDecider::MakeObjectDecision(const PathData &path_data,
                                      const std::string &blocking_obstacle_id,
                                      PathDecision *const path_decision) {
-  DCHECK_NOTNULL(path_decision);
+  CHECK_NOTNULL(path_decision);
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_decider/path_decider.cc]: ditto

File: modules/planning/tasks/deciders/path_decider/path_decider.cc [hunk 12]
```diff
@@ -82,7 +82,7 @@ bool PathDecider::MakeStaticObstacleDecision(
     const PathData &path_data, const std::string &blocking_obstacle_id,
     PathDecision *const path_decision) {
   // Sanity checks and get important values.
-  DCHECK_NOTNULL(path_decision);
+  CHECK_NOTNULL(path_decision);
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_decider/path_decider.cc]: ditto

**Pull request merged**